File: ABC/ABC435/c++/problemB.cpp
#include <bits/stdc++.h>

using namespace std;

#define rep(i,n) for(int i = 0; i < n; i++)
#define v(i) vector<i>

bool yakusu(v(int) A, int i, int j){
	int sum = 0;
	for(int k = i; k <= j; k++){
		sum += A[k];
	}

	for(int k = i; k <= j; k++){
		if(sum % A[k] == 0){
			return true;
		}
	}

	return false;

}

int main(){
	int N;
	cin >> N;
	v(int) A(N);
	rep(i, N){ cin >> A[i]; }

	int res = 0;

	rep(i,N){
		for(int j = i; j < N; j++){
			if(j > i && (yakusu(A, i, j) == false)){
				res++;
			}
		}
	}
	cout << res;
	return 0;
}

File: ABC/ABC435/python/problemB.py

def yakusu(A, i, j):
    sum = 0

    for k in range(i, j+1):
        sum += A[k]

    for k in range(i, j+1):
        if(sum % A[k] == 0):
            return True
        
    return False

def main():
    N = int(input())
    A = list(map(int, input().split()))

    res = 0

    for i in range(N):
        for j in range(i,N):
            if((j > i) and (yakusu(A,i,j) == False)):
                res += 1

    print(res)

if __name__ == "__main__":
    main()